### PR TITLE
DDF-3330 - Shape File Transformer

### PR DESCRIPTION
**Pull request by dcruver**
#### What does this PR do?
Added catalog-transformer-shapefilereader and catalog-transformer-shapefile bundles.

#### Who is reviewing it? 
@gordocanchola 
@beyelerb 
@kcwire 

#### Select relevant component teams: 

https://github.com/orgs/codice/teams

#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)

@jlcsmith
@pklinef

#### How should this be tested? (List steps with links to updated documentation)
1) Ingest a shapefile
2) In Intrigue, search for that shapefile
    - verify that the search results contain the ingested file
3) Select the Inspector details tab
    - verify that the metacard contains shapefile-specific attributes
4) Select the 3D View
5) Select the shapefile item in the search results pane
    - verify that the shapefile location is represented on the globe

#### What are the relevant tickets?

[DDF-3330](https://codice.atlassian.net/browse/DDF-3330)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests
#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/api/ShapeFileAttributes.java [hunk 1]
```diff
@@ -0,0 +1,95 @@
...
+public class ShapeFileAttributes implements ShapeFile, MetacardType {
+  private static final Set<AttributeDescriptor> DESCRIPTORS;
+
+  private static final String NAME = "shapefile";
+
+  static {
+    Set<AttributeDescriptor> descriptors = new HashSet<>();
+    descriptors.add(
+        new AttributeDescriptorImpl(
+            SHAPE_MIN_LAT,
```
Comment (bdeining) [inline catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/api/ShapeFileAttributes.java]: Why do we define a MetacardType for shape files and add min/max lat/lon attributes?  The location attribute isn't sufficient for this?

Comment (dcruver) [reply catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/api/ShapeFileAttributes.java]: Shapefiles are typically very complex. They follow a river or the borders of a city. So these attributes make it easier to display a bounding box that encompasses the shape on the 3D view.

**Comment by dcruver** [comment]
build now

File: catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/ShapeFileInputTransformer.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+  /**
+   * @param input input stream containing the shape file
+   * @param id an id which can be used to set the metacard id
+   * @return the metacard created from the shape file
+   */
+  private Metacard transformShapeFile(InputStream input, String id)
+      throws CatalogTransformerException {
+    MetacardImpl metacard = new MetacardImpl(metacardType);
+
+    if (id != null) {
```
Comment (jlcsmith) [inline catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/ShapeFileInputTransformer.java]: ✏️ nit, probably should just use `StinrgUtils.isNotBlank()` https://commons.apache.org/proper/commons-lang/javadocs/api-2.6/org/apache/commons/lang/StringUtils.html#isNotBlank(java.lang.String)

File: catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/ShapeFileInputTransformer.java [hunk 3]
```diff
@@ -0,0 +1,120 @@
...
+    metacard.setAttribute(Topic.CATEGORY, SHAPE_CATAGORY);
+
+    try {
+      ShapeFileReader reader = factory.createShapeFileReader(input);
+      reader.readHeader();
+
+      ShapeFileHeader header = reader.getShapeFileHeader();
+
+      metacard.setAttribute(ShapeFile.SHAPE_TYPE, header.getShapeType().toString());
+      metacard.setLocation(header.getBbox());
```
Comment (jlcsmith) [inline catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/ShapeFileInputTransformer.java]: we should be using the actual geo (or an approximation ) from the shapefile

File: catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/ShapeFileInputTransformer.java [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+
+      metacard.setAttribute(ShapeFile.SHAPE_MIN_LAT, header.getMinLat());
```
Comment (jlcsmith) [inline catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/ShapeFileInputTransformer.java]: I don't see a need to retain the min/max lat/lon in the metacard

File: catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/ShapeFileInputTransformer.java [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+      metacard.setAttribute(ShapeFile.SHAPE_MAX_LAT, header.getMaxLat());
+      metacard.setAttribute(ShapeFile.SHAPE_MIN_LON, header.getMinLon());
+      metacard.setAttribute(ShapeFile.SHAPE_MAX_LON, header.getMaxLon());
+    } catch (Exception e) {
```
Comment (jlcsmith) [inline catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/ShapeFileInputTransformer.java]: Is there a reason  to catch Exception or can we catch specific exceptions instead?

File: catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/api/ShapeFile.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.transformer.shape.file.transformer.api;
+
+/** This interface defines the extra metacard taxonomy types a shape file has. */
+public interface ShapeFile {
```
Comment (jlcsmith) [inline catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/api/ShapeFile.java]: Again,. I don't think we need these attributes

File: catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/api/ShapeFileAttributes.java (hunk 1 above)
Comment (jlcsmith) [reply catalog/transformer/catalog-transformer-shapefile/src/main/java/ddf/catalog/transformer/shape/file/transformer/api/ShapeFileAttributes.java]: Hoping we can remove these and try to do a smart approximation of the geo in the shapefile and set it on the location attribute

**Comment by rzwiefel** [comment]
@dcruver Do we just need reviews on this to get it merged? Or should we close until it is prioritized again?

**Comment by coyotesqrl** [comment]
@dcruver Tagging again to ask on the status of this PR. Should it be updated and completed so it can be merged, or should it be closed at this time?

**Comment by dcruver** [comment]
This has been on-hold through 2 releases and has become stale. Closing.

**Pull request closed without merging**